the-collab-lab/tcl-27-smart-shopping-list
# Generate unique token and redirect to /list

## Description
* In a handleClick function, we generated a unique token using getToken from token.js and stored it in localStorage
* Used useState to set loggedIn state to true
* Based on loggedIn state, redirect to List component
* If token exists in localStorage, automatically redirect to List component when app renders

## Related Issue
Closes #3

## Acceptance Criteria
[Acceptance Criteria](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/3)

* For users who do not already have a token/list, a button or link exists on the home screen that allows them to create a new list
* Clicking the button/link generates a new token and saves it to localStorage
* Once the token has been created and saved, the user is redirected to the "list" view
* For users who do have a token already saved in localStorage, they should be shown/redirected to the "list" view

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |


## Testing Steps / QA Criteria
From your terminal, pull down this branch with git pull origin sk-tc-new-shopping-list and check that branch out with git checkout sk-tc-new-shopping-list.  Then npm install or yarn to install the newly added dependency locally and npm start or yarn start to launch the app.  

Click on the "Create List..." button to generate a new token, which is stored in localStorage. Make sure to check that the app redirects to list view and that the url route changes to '/list'. Also, without clearing localStorage, refresh the app from the root url and make sure that it automatically renders list view.
